src/main/java/org/terasology/launcher/gui/javafx/ApplicationController.java

```diff
@@ -504,6 +523,56 @@ private void initComboBoxes() {
         });
     }
 
+    private void selectJobBoxItemForJob(final ComboBox<PackageItem> jobBox, final String jobId) {
+        PackageItem packageItem = jobBox.getItems().stream()
+                .filter(item -> item.versionItems.stream()
+                        .anyMatch(vItem -> vItem.linkedPackageProperty.get().getId().equals(jobId)))
+                .findFirst()
+                .orElseGet(() -> null);
+
+        if (packageItem != null) {
+            jobBox.getSelectionModel().select(packageItem);
+        } else {
+            jobBox.getSelectionModel().select(0);
+        }
+    }
+
+    private void selectBuildVersionBoxItemForVersion(final ComboBox<VersionItem> buildVersionBox, final String version) {
+        VersionItem versionItem = buildVersionBox.getItems().stream()
+                .filter(item -> item.versionProperty.get().equals(version))
+                .findFirst()
+                .orElseGet(() -> null);
+
+        if (versionItem != null) {
+            buildVersionBox.getSelectionModel().select(versionItem);
+        } else {
+            buildVersionBox.getSelectionModel().select(0);
+        }
+    }
+
+    private void initializeComboBoxSelection() {
```

Comment:
It would be really nice to reduce the boilerplate code here to bring the logic more into the foreground. With Java9 and later (see https://docs.oracle.com/javase/9/docs/api/java/util/Optional.html#or-java.util.function.Supplier-) (or a another functional library...) we could chain optionals to achieve that:

```java
Optional<Package> pkg = getLastPlayed()
	.or(() -> getLastInstalled())
	.or(() -> getLatestInstalled());

this.select(pkg);
```

Reply:
I would appreciate a `//TODO:` comment about this so that we don't forget about it later.